### PR TITLE
range-v3: init at 2017-01-30

### PR DESCRIPTION
**Pull request by d-xo**
###### Motivation for this change

Add the range-v3 c++ library

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

As a side note I decided not to run the test suite was that the correct choice?

### CODE REVIEW

**Comment by xwvvvvwx** [comment]
`nox-review` fails with, I don't really understand why, pretty sure that this is unrelated to my change?

```
nix-shell -p nox --run "nox-review wip --against c3fdfb0df339054f65df140b4da32ff4b01ce18a"
==> We're in a git repo, trying to fetch it
error: while evaluating ‘callPackageWith’ at /home/david/.nox/nixpkgs/lib/customisation.nix:95:35, called from /home/david/.nox/nixpkgs/pkgs/top-level/all-packages.nix:11401:17:
while evaluating ‘makeOverridable’ at /home/david/.nox/nixpkgs/lib/customisation.nix:54:24, called from /home/david/.nox/nixpkgs/lib/customisation.nix:99:8:
while evaluating anonymous function at /home/david/.nox/nixpkgs/pkgs/os-specific/linux/phc-intel/default.nix:1:1, called from /home/david/.nox/nixpkgs/lib/customisation.nix:56:12:
attribute ‘features’ missing, at /home/david/.nox/nixpkgs/pkgs/os-specific/linux/phc-intel/default.nix:7:9
Traceback (most recent call last):
  File "/nix/store/7am4pygv9bmhj2h4rzh6rlwrpvsihl2f-nox-0.0.5/bin/.nox-review-wrapped", line 12, in <module>
    sys.exit(cli())
  File "/nix/store/d88wdn74mq0rks901snnrw8javbgs9wm-python3.5-click-6.6/lib/python3.5/site-packages/click/core.py", line 716, in __call__
    return self.main(*args, **kwargs)
  File "/nix/store/d88wdn74mq0rks901snnrw8javbgs9wm-python3.5-click-6.6/lib/python3.5/site-packages/click/core.py", line 696, in main
    rv = self.invoke(ctx)
  File "/nix/store/d88wdn74mq0rks901snnrw8javbgs9wm-python3.5-click-6.6/lib/python3.5/site-packages/click/core.py", line 1060, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/nix/store/d88wdn74mq0rks901snnrw8javbgs9wm-python3.5-click-6.6/lib/python3.5/site-packages/click/core.py", line 889, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/nix/store/d88wdn74mq0rks901snnrw8javbgs9wm-python3.5-click-6.6/lib/python3.5/site-packages/click/core.py", line 534, in invoke
    return callback(*args, **kwargs)
  File "/nix/store/d88wdn74mq0rks901snnrw8javbgs9wm-python3.5-click-6.6/lib/python3.5/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/nix/store/7am4pygv9bmhj2h4rzh6rlwrpvsihl2f-nox-0.0.5/lib/python3.5/site-packages/nox/review.py", line 81, in wip
    attrs = differences(packages_for_sha(sha),
  File "/nix/store/y69jandw5yg884wssl4pn0zp8hkpsjnk-python3.5-dogpile.cache-0.5.4/lib/python3.5/site-packages/dogpile/cache/region.py", line 1013, in decorate
    should_cache_fn)
  File "/nix/store/y69jandw5yg884wssl4pn0zp8hkpsjnk-python3.5-dogpile.cache-0.5.4/lib/python3.5/site-packages/dogpile/cache/region.py", line 640, in get_or_create
    async_creator) as value:
  File "/nix/store/244hvg0mjb2a4hbvw5jvmkgccymgcl50-python3.5-dogpile.core-0.4.1/lib/python3.5/site-packages/dogpile/core/dogpile.py", line 158, in __enter__
    return self._enter()
  File "/nix/store/244hvg0mjb2a4hbvw5jvmkgccymgcl50-python3.5-dogpile.core-0.4.1/lib/python3.5/site-packages/dogpile/core/dogpile.py", line 98, in _enter
    generated = self._enter_create(createdtime)
  File "/nix/store/244hvg0mjb2a4hbvw5jvmkgccymgcl50-python3.5-dogpile.core-0.4.1/lib/python3.5/site-packages/dogpile/core/dogpile.py", line 149, in _enter_create
    created = self.creator()
  File "/nix/store/y69jandw5yg884wssl4pn0zp8hkpsjnk-python3.5-dogpile.cache-0.5.4/lib/python3.5/site-packages/dogpile/cache/region.py", line 612, in gen_value
    created_value = creator()
  File "/nix/store/y69jandw5yg884wssl4pn0zp8hkpsjnk-python3.5-dogpile.cache-0.5.4/lib/python3.5/site-packages/dogpile/cache/region.py", line 1009, in creator
    return fn(*arg, **kw)
  File "/nix/store/7am4pygv9bmhj2h4rzh6rlwrpvsihl2f-nox-0.0.5/lib/python3.5/site-packages/nox/nixpkgs_repo.py", line 90, in packages_for_sha
    return packages(repo.path)
  File "/nix/store/7am4pygv9bmhj2h4rzh6rlwrpvsihl2f-nox-0.0.5/lib/python3.5/site-packages/nox/nixpkgs_repo.py", line 81, in packages
    universal_newlines=True)
  File "/nix/store/42p7bp41vr94bb8ip0nq0v91fcb8ixmr-python3-3.5.2/lib/python3.5/subprocess.py", line 626, in check_output
    **kwargs).stdout
  File "/nix/store/42p7bp41vr94bb8ip0nq0v91fcb8ixmr-python3-3.5.2/lib/python3.5/subprocess.py", line 708, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['nix-env', '-f', '/home/david/.nox/nixpkgs', '-qaP', '--out-path', '--show-trace']' returned non-zero exit status 1
```

File: pkgs/development/libraries/range-v3/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{ stdenv, fetchFromGitHub, cmake }:
```
Comment (joachifm) [inline pkgs/development/libraries/range-v3/default.nix]: `cmake` looks unused

Comment (xwvvvvwx) [inline pkgs/development/libraries/range-v3/default.nix]: thanks! fixed :)

**Comment by joachifm** [comment]
@xwvvvvwx That error is fixed in current master, it's assuredly unrelated to your PR.

Have you considered adding yourself as the maintainer of this package?

**Comment by xwvvvvwx** [comment]
@joachifm done :)

**Comment by joachifm** [comment]
Thank you

**Pull request merged**